File: android/DKMadsAdCachePolicy.kt
package com.dkmads.ssp

import java.util.Date
import java.util.concurrent.TimeUnit

/** In-memory ad cache TTL aligned with [sdk/spec/video-lifecycle-v1.md]. */
object DKMadsAdCachePolicy {
    val FULLSCREEN_MAX_AGE_MS = TimeUnit.HOURS.toMillis(4)
    val INLINE_MAX_AGE_MS = TimeUnit.HOURS.toMillis(1)

    fun maxAgeMs(format: AdFormat): Long = when (format) {
        AdFormat.BANNER, AdFormat.NATIVE -> INLINE_MAX_AGE_MS
        else -> FULLSCREEN_MAX_AGE_MS
    }

    fun isExpired(loadedAt: Date?, format: AdFormat, now: Date = Date()): Boolean {
        if (loadedAt == null) return true
        return now.time - loadedAt.time > maxAgeMs(format)
    }
}

File: ios/Sources/DKMadsSSPSDK/Core/DKMadsAdCachePolicy.swift
import Foundation

/// In-memory ad cache TTL aligned with `sdk/spec/video-lifecycle-v1.md`.
enum DKMadsAdCachePolicy {
    static let fullscreenMaxAge: TimeInterval = 4 * 3600
    static let inlineMaxAge: TimeInterval = 3600

    static func maxAge(for format: AdFormat) -> TimeInterval {
        switch format {
        case .banner, .native:
            return inlineMaxAge
        case .interstitial, .rewarded, .splash, .video, .audio:
            return fullscreenMaxAge
        }
    }

    static func isExpired(loadedAt: Date?, format: AdFormat, now: Date = Date()) -> Bool {
        guard let loadedAt else { return true }
        return now.timeIntervalSince(loadedAt) > maxAge(for: format)
    }
}
